### DISCUSSION TITLE
Z axis end stop

### DISCUSSION THREAD

**Discussion by wfishrn** [arkypita/LaserGRBL#938] [Q&A]
Only running 2 axis, X and Y, grbl set to home ZYX in that order, so homing fails, any ideas on how to disable the Z homing or can I some how do a work around, Im even thinking of using a momentary switch if needed to fool the machine into thinking there is a Z axes??
Thank You

**Comment by gmmanonymus111** [comment]
You need to recompile GRBL to disable Z homing AFAIK.
To do so, edit config.h in GRBL's folder:
Replace
```
#define HOMING_CYCLE_0 (1<<Z_AXIS)
#define HOMING_CYCLE_1 ((1<<X_AXIS)|(1<<Y_AXIS))
```
with
```
#define HOMING_CYCLE_0 (1<<X_AXIS)
#define HOMING_CYCLE_1 (1<<Y_AXIS)
```
to do X homing first, and Y next, or
```
#define HOMING_CYCLE_0 ((1<<X_AXIS)|(1<<Y_AXIS))
```
to do X and Y homing at the same time.
After editing, you can flash it with Arduino IDE.

**Comment by wfishrn** [comment]
Ok will this affect the laser connected to the Z+ for PWM control, just thought of that.. ???

**Comment by StuartB4** [comment]
No. I assume you are using a Protoneer CNC shield and UNO.
I do it the same way. X,Y home, pwm wire for laser connected to Z+/- on shield.
Works fine.
